pkg/distributor/distributor.go

```diff
@@ -728,15 +726,15 @@ func (d *Distributor) prePushHaDedupeMiddleware(next push.Func) push.Func {
 
 		removeReplica, err := d.checkSample(ctx, userID, cluster, replica)
 		if err != nil {
-			if errors.Is(err, replicasNotMatchError{}) {
+			var replicasNotMatchErr distributorerror.ReplicasNotMatch
+			if errors.As(err, &replicasNotMatchErr) {
```

Comment:
Since  you're still pushing changes, nit, this would be shorter and no need to pollute with another variable:
```suggestion
			if errors.As(err, &distributorerror.ReplicasNotMatch{}) {
```